android/src/main/java/com/swmansion/reanimated/layoutReanimation/AnimationsManager.java

```diff
@@ -600,6 +600,9 @@ private void maybeDropAncestors(View exitingView) {
 
   private void removeView(View view, @Nullable ViewGroup parent) {
     int tag = view.getId();
+    if (tag == -1) {
+      return;
+    }
```

Comment:
Good job with debugging!
I wondering if we safety can just stop function here, instead of call `mReanimatedNativeHierarchyManager.publicDropView(view);` and `parent.removeView(view);` Can we be sure that this changes will not drove to memory leaks?

Reply:
I've replaced `return` with your code and tested with `Memory Profiler`. I haven't seen any leaks:
![Screenshot 2023-05-16 at 10 24 26](https://github.com/software-mansion/react-native-reanimated/assets/56199675/023099f5-e841-4679-b4ff-6a2c873d133e)

Reply:
I've checked that `mCallbacks.put(tag, callback);` is not called for `tag=-1`

Reply:
I saw that you reverted changes with view removing. Let's try again 😄
Which call causes a crash?
`mReanimatedNativeHierarchyManager.publicDropView(view)` or  `parent.removeView(view)` ?

It seems that mReanimatedNativeHierarchyManager.publicDropView can be skipped, however, parent.removeView is still required. But you need to check it to be sure.

Reply:
Yes, when you move the condition to skip only `parent.removeView(view)` the bug does not appear.